### ISSUE TITLE
qrScanner.stopCapture() is not working

### ISSUE THREAD

**Issue by frabrunelle** [hitchcott/meteor-qr-code-scanner#9]
Is it just me or the `qrScanner.stopCapture()` function is not working?

The webcam stays on, the camera icon keeps saying "This page is accessing your camera.", the canvas with the video feed is still there.

![screen shot 2015-10-18 at 2 06 59 am](https://cloud.githubusercontent.com/assets/1114844/10562808/f257c3c2-753c-11e5-8bbf-2e7dbd77c6e5.png)

![screen shot 2015-10-18 at 2 05 28 am](https://cloud.githubusercontent.com/assets/1114844/10562807/ceda9942-753c-11e5-828b-530f341b4b16.png)

It would be nice to have this working :smiley: 

My solution for now is to call `location.reload()` in order to reload the page but I would prefer to be able to use `qrScanner.stopCapture()`.

**Comment by rohanray** [comment]
Hi @frabrunelle I had the same issue. Try using `qrScanner.off('scan');` instead of `qrScanner.stopCapture()`. Let me know if this works.

**Comment by frabrunelle** [comment]
@rohanray: I just tried using `qrScanner.off('scan');` but it doesn't work as well :disappointed: 

If you were able to get it working, could you show me a code example or demo app? Thank you for your help!

**Comment by rohanray** [comment]
Hi @frabrunelle 

This is the template code:

```
<template name="scanQR">

            scanning QR
            {{> qrScanner}}

</template>
```

This is the js 

```
qrScanner.on('scan', function(err, message) {
    if (message !== null) {
       console.log(message);
       qrScanner.off('scan');
        return;
    }
});

Template.scanQR.onDestroyed(function () {
qrScanner.off('scan');
});
```

Let me know if this helps. If not, I will try to create a gist tomorrow.

**Comment by IstoraMandiri** [comment]
`off('scan')` will only stop the events from firing, but will continue to update the video

Sorry I won't be able to fix this immediately, I simply tested PR and it seemed to work for me. Happy to accept a fix if someone can figure out what's going on.

FYI `stopCapture` looks like this:

```
stopCapture = ->
  if localMediaStream
    try
      localMediaStream.stop()
      localMediaStream.active = false
  if localMediaInterval
    Meteor.clearInterval localMediaInterval
  showingCanvas = false
  qrReactiveDict.set 'message', null
  qrScanner.off 'scan'
```

@rohanry Does `stopCapture` still faill with `hitchcott:qr-scanner` ?

**Comment by rohanray** [comment]
@hitchcott 
qrScanner.stopCapture(); is indeed working with hitchcott:qr-scanner :smiley: 

@frabrunelle Can you share your code and output of meteor list?

**Comment by frabrunelle** [comment]
> qrScanner.stopCapture(); is indeed working with hitchcott:qr-scanner :smiley:

@rohanray: For me it's not working. The camera continues to be active. I would like to see the red circle disappear.

I just made a very [simple demo](https://github.com/frabrunelle/meteor-qr-scanner-demo) that shows the issue.

If you are able to make it work, it would be nice if you could build a sample application and upload it to GitHub so that I could test it myself. Thank you.

I would also like a way to reactivate the camera after it has been deactivated. So there should not only be `qrScanner.stopCapture()` but also `qrScanner.startCapture()`.

I found [this library](https://github.com/dwa012/html5-qrcode) which seems to have support this feature. I haven't tried it yet, but I am thinking of creating a Meteor wrapper for it if it does work.

> When you're done using the camera, call $('#reader').html5_qrcode_stop() to stop scanning for QR codes, close the stream, and relinquish permissions granted by the user.
> (This will result in the "camera on" light on their device turning off)

**Comment by rohanray** [comment]
@frabrunelle I cloned your repo and it's working fine for me. The "red circle" indicating camera disappears after closing the alert box displaying the scanned message. 

Might be a browser/environment dependency?

Mine is as below: 
Windows 10 (64 bit), Chrome Version 46.0.2490.71 m (64-bit)

**Comment by frabrunelle** [comment]
I'm on OS X El Capitan Version 10.11.1 running Google Chrome Version 47.0.2526.27 beta (64-bit).

I haven't tried on the stable channel but I think in might be due to recent changes in the MediaStream API.

https://developers.google.com/web/updates/2015/07/mediastream-deprecations?hl=en

> Use MediaStreamTrack.stop() to stop streaming, not MediaStream.stop().

Right now in the code we do `localMediaStream.stop()` ([line 49](https://github.com/hitchcott/meteor-qr-code-scanner/blob/master/qr-scanner.coffee#L49)). But it seems like we should be doing`localMediaStream.getTracks()[0].stop()` instead. 

> Removing stop() from MediaStream did not remove any real functionality: processes for detaching source devices and so on have to be done on MediaStreamTrack anyway. Use stop() on MediaStreamTrack instead:

```
navigator.getUserMedia({audio: false, video: true},
    function(stream) {
         // can also use getAudioTracks() or getVideoTracks()
        var track = stream.getTracks()[0];  // if only one media track
        // ...
        track.stop();
    },
    function(error){
        console.log('getUserMedia() error', error);
    });
```

Relevant issues: 
- [How do you stop the video from scanning?](https://github.com/dwa012/html5-qrcode/issues/23)
- [Change deprecated call to MediaStream.stop() to MediaStreamTrack.stop()](https://github.com/dwa012/html5-qrcode/pull/26)
- [Fix deprecation warning on MediaStream.stop().](https://github.com/dwa012/html5-qrcode/pull/27)

Now it makes sense why it wasn't working for me: it's because `MediaStream.stop()` is being deprecated in Chrome 47, which is the version I am currently using.

> 'MediaStream.stop()' is deprecated and will be removed in M47, around November 2015. Please use 'MediaStream.active' instead.

**Comment by frabrunelle** [comment]
I just submitted a PR (#10) which fixes the issue :smile:

**Comment by rohanray** [comment]
That's really good to hit the spot! Waiting for your PR :+1:

**Comment by frabrunelle** [comment]
@rohanray: My PR is [here](https://github.com/hitchcott/meteor-qr-code-scanner/pull/10) :wink:

**Comment by rohanray** [comment]
thanks @frabrunelle Cheers!

**Comment by IstoraMandiri** [comment]
Fixed in 1.0.1
